lib/resty/kafka/protocol/consumer.lua

```diff
@@ -0,0 +1,257 @@
+local protocol = require "resty.kafka.protocol.common"
+local proto_record = require "resty.kafka.protocol.record"
+local request = require "resty.kafka.request"
+
+local ffi = require "ffi"
+local table_insert = table.insert
+
+local _M = {}
+
+
+_M.LIST_OFFSET_TIMESTAMP_LAST  = -1
+_M.LIST_OFFSET_TIMESTAMP_FIRST = -2
+_M.LIST_OFFSET_TIMESTAMP_MAX   = -3
+
+
+local function _list_offset_encode(req, isolation_level, topic_partitions)
+    req:int32(-1) -- replica_id
+
+    if req.api_version >= protocol.API_VERSION_V2 then
+        req:int8(isolation_level) -- isolation_level
+    end
+
+    req:int32(topic_partitions.topic_num)   -- [topics] array length
+
+    for topic, partitions in pairs(topic_partitions.topics) do
+        req:string(topic) -- [topics] name
+        req:int32(partitions.partition_num) -- [topics] [partitions] array length
+
+        for partition_id, partition_info in pairs(partitions.partitions) do
+            req:int32(partition_id) -- [topics] [partitions] partition_index
+            req:int64(ffi.new("int64_t", partition_info.timestamp)) -- [topics] [partitions] timestamp
```

Comment:
we don't need `ffi.new` here? to avoid creating a new cdata.

Reply:
It requires a real `int64` data, otherwise it will repeat the `int32` data twice in the general case. Here a example:

```text
input: int32 (100) - default lua integer
encoded: 00000000 00000000 00000000 01100100 00000000 00000000 00000000 01100100

input: int64 (100) - int64 cdata
encoded: 00000000 00000000 00000000 00000000 00000000 00000000 00000000 01100100
```

Is there a way we can add an `LL` suffix to a variable? I'm not familiar with that.

Reply:
oh, seems it's a bug in the request.int64.
we can call `int * 1LL` in `str_int64 `.

Reply:
fix, btw, Sorry for I clicked on the wrong one to force push (I turned off the confirm before) 😳